### PR TITLE
Move preflight deps behind build tags

### PR DESCRIPTION
**Pull request by jordan-rash**

### CODE REVIEW

File: internal/node/preflight/init.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/fatih/color"
+)
+
+var (
+	VERSION = "development"
```
Comment (jordan-rash) [inline internal/node/preflight/init.go:13]: This needs to be updated at build time

File: internal/node/preflight/preflight.go [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/synadia-io/nex/internal/models"
+	"github.com/synadia-io/nex/internal/node/templates"
+)
+
+var (
+// TODO: move this to flag
```
Comment (jordan-rash) [inline internal/node/preflight/preflight.go:25]: clean up

File: internal/node/preflight/preflight.go [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+	for _, r := range required {
+		if r.satisfied && !config.ForceDepInstall {
+			continue
+		}
+
+		var input []byte
+		var err error
+
+		if !config.ForceDepInstall {
+			fmt.Printf("⛔ You are missing required dependencies for [%s], do you want to install? [y/N] ", red(r.description))
```
Comment (jordan-rash) [inline internal/node/preflight/preflight.go:94]: use helper

File: internal/node/preflight/preflight.go [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+
+var (
+// TODO: move this to flag
+)
+
+type requirement struct {
+	name        string
+	path        []string
+	description string
+	nosandbox   bool
```
Comment (jordan-rash) [inline internal/node/preflight/preflight.go:32]: i dont think this field it needed

**Review by autodidaddict — APPROVED** [review]

File: internal/node/preflight/preflight.go (hunk 4 above)
Comment (jordan-rash) [reply internal/node/preflight/preflight.go:32]: im wrong

File: spec/node_linux_test.go [hunk 5]
```diff
@@ -155,11 +155,15 @@ var _ = Describe("nex node", func() {
 					Context("when the specified default_resource_dir does not exist on the host", func() {
 						BeforeEach(func() {
 							nodeConfig.DefaultResourceDir = filepath.Join(os.TempDir(), fmt.Sprintf("%d-non-existent-nex-resource-dir", _fixtures.seededRand.Int()))
+							_, err := os.Stat(nodeConfig.DefaultResourceDir)
+							Expect(err).To(Equal(os.ErrNotExist))
 						})
 
-						It("should [not] return an error", func(ctx SpecContext) {
+						It("should create the default_resource_dir", func(ctx SpecContext) {
```
Comment (kthomas) [inline spec/node_linux_test.go:161]: 🥺 the amount of joy this commit has brought me

**Review by kthomas — APPROVED** [review]
Looks like the ~~linter is~~ tests are still unhappy, but this looks great otherwise.

**Review by kthomas — DISMISSED** [review]

**Review by autodidaddict — APPROVED** [review]

**Pull request merged**